File: src/main/java/upbrella/be/store/repository/StoreDetailRepositoryCustom.java
package upbrella.be.store.repository;

import upbrella.be.store.dto.response.SingleStoreResponse;

import java.util.List;
import java.util.Optional;
import upbrella.be.store.entity.StoreDetail;

public interface StoreDetailRepositoryCustom {

    List<StoreDetail> findAllStores();

    Optional<StoreDetail> findByStoreMetaIdUsingFetchJoin(long storeMetaId);

    List<SingleStoreResponse> findAllStoresForAdmin();
}

File: src/main/kotlin/upbrella/be/store/repository/StoreDetailRepositoryCustom.kt
package upbrella.be.store.repository

import upbrella.be.store.dto.response.SingleStoreResponse
import upbrella.be.store.entity.StoreDetail
import java.util.Optional

interface StoreDetailRepositoryCustom {
    fun findAllStores(): List<StoreDetail>

    fun findByStoreMetaIdUsingFetchJoin(storeMetaId: Long): Optional<StoreDetail>

    fun findAllStoresForAdmin(): List<SingleStoreResponse>
}